rust/compiler/src/parser/mod.rs

```diff
@@ -231,11 +235,14 @@ pub fn decl_or_annotation(i: Input) -> Res<Input, DeclOrAnnotation> {
 pub fn decl(i: Input) -> Res<Input, adlast::Decl<TypeExpr0>> {
     let (i, annotations) = many0(prefix_annotation)(i)?;
     let (i, (name, dtype)) = decl_type(i)?;
-
+    let ma = merge_annotations(annotations);
```

Comment:
If the signature of merge_annotations is as suggested below, this:

```
    let ma = merge_annotations(annotations);
    if let Err(e) = ma {
        return Err(custom_error(i, e.to_string()));
    }
    let decl = adlast::Decl {
        name: name.to_owned(),
        r#type: dtype,
        annotations: ma.unwrap(),
        version: Maybe::nothing(),
    };
```

can become:

```
    let annotations = merge_annotations(annotations).map_err(|emsg| custom_error(i, emsg))?;
    let decl = adlast::Decl {
        name: name.to_owned(),
        r#type: dtype,
        annotations,
        version: Maybe::nothing(),
    };
```

Reply:
This is a common pattern:
     - using the ? operator to automatically propagate errors
     - using `Result::map_err()` to convert an error from a callee to one expected by the caller.